Firestore/core/src/firebase/firestore/util/firebase_assert.h

```diff
@@ -103,6 +103,11 @@
   FIREBASE_DEV_ASSERT_MESSAGE_WITH_EXPRESSION(expression, expression, \
                                               __VA_ARGS__)
 
+// Indicates an area of the code that cannot be reached (except possibly due to
+// undefined behaviour). The only reasonable thing to do in these cases is to
```

Comment:
Nit: IIUC, there is no undefined behavior involved in this case. From the thread you linked:

> Enumerations are simply named constants, they don't affect the type, or codegen, or what the valid values of the type are (except where the values of the highest and lowest determine the range of valid values).

I'll slightly expand the example from the thread:
```cpp
enum class Foo { SomeValue = 1, }; // Underlying type is int
std::string Bar() {
  Foo foo = static_cast<Foo>(42); // This is well-defined
  switch (foo) {
  case SomeValue:
    return "foobar";
  }
  // Control reaches here, resulting in undefined behavior
}
```

Reply:
I've fixed up the comment slightly.